python/grass/jupyter/interactivemap.py

```diff
@@ -347,13 +348,139 @@ def add_layer_control(self, **kwargs):
 
+    def draw_computational_region(self):
+        import ipywidgets as widgets
+        from ipyleaflet import Rectangle, WidgetControl
+
+        region_mode_button = widgets.ToggleButton(
+            icon="square",
+            value=False,
+            tooltip="Click to add computational region",
+            button_style="info",
+            layout=widgets.Layout(width="33px", margin="0px 0px 0px 0px"),
+        )
+
+        save_button = widgets.Button(
+            description="Save Region",
+            button_style="success",
+            layout=widgets.Layout(width="100px", margin="0px 0px 0px 5px"),
+            disabled=True,
+        )
+
+        bottom_output_widget = widgets.Output(
+            layout={
+                "width": "100%",
+                "max_height": "300px",
+                "max_width": "300px",
+                "overflow": "auto",
+                "display": "none",
+            }
+        )
+
+        region_coordinates = {}
+        rectangle = None
+        save_button_control = None
+
+        def update_output():
+            with bottom_output_widget:
+                bottom_output_widget.clear_output()
+                print("Current Bounds:")
+                print(f"North: {region_coordinates['North']}")
+                print(f"South: {region_coordinates['South']}")
+                print(f"East: {region_coordinates['East']}")
+                print(f"West: {region_coordinates['West']}")
+
+        def on_rectangle_change(event, property_name, value):
+            nonlocal region_coordinates
+            if property_name == "bounds":
+                latlon_bounds = value
+                region_coordinates = {
+                    "North": latlon_bounds[1][0],
+                    "South": latlon_bounds[0][0],
+                    "East": latlon_bounds[1][1],
+                    "West": latlon_bounds[0][1],
+                }
+                update_output()
+
+        def toggle_region_mode(change):
+            nonlocal rectangle, save_button_control, region_coordinates
+
+            if change["new"]:
+                if rectangle is None:
+                    latlon_bounds = get_computational_region_bb()
+                    region_coordinates = {
+                        "North": latlon_bounds[1][0],
+                        "South": latlon_bounds[0][0],
+                        "East": latlon_bounds[1][1],
+                        "West": latlon_bounds[0][1],
+                    }
+                    rectangle = Rectangle(
+                        bounds=latlon_bounds,
+                        color="red",
+                        fill_color="red",
+                        fill_opacity=0.5,
+                        draggable=True,
+                        editable=True,
+                        transform=True,
+                        name="computational_region",
+                    )
+                    rectangle.observe(on_rectangle_change, names="bounds")
```

Comment:
"bounds" is not working for me, try "locations"

Reply:
I tried it out, but the widget is still not getting updates.

Reply:
OK, here is the diff, with theses changes the widget is getting updates:
```
+++ b/python/grass/jupyter/interactivemap.py
@@ -390,17 +390,16 @@ class InteractiveMap:
                 print(f"East: {region_coordinates['East']}")
                 print(f"West: {region_coordinates['West']}")
 
-        def on_rectangle_change(event, property_name, value):
+        def on_rectangle_change(value):
             nonlocal region_coordinates
-            if property_name == "bounds":
-                latlon_bounds = value
-                region_coordinates = {
-                    "North": latlon_bounds[1][0],
-                    "South": latlon_bounds[0][0],
-                    "East": latlon_bounds[1][1],
-                    "West": latlon_bounds[0][1],
-                }
-                update_output()
+            latlon_bounds = value["new"][0]
+            region_coordinates = {
+                "North": latlon_bounds[2]['lat'],
+                "South": latlon_bounds[0]['lat'],
+                "East": latlon_bounds[2]['lng'],
+                "West": latlon_bounds[0]['lng'],
+            }
+            update_output()
 
         def toggle_region_mode(change):
             nonlocal rectangle, save_button_control, region_coordinates
@@ -424,7 +423,7 @@ class InteractiveMap:
                         transform=True,
                         name="computational_region",
                     )
-                    rectangle.observe(on_rectangle_change, names="bounds")
+                    rectangle.observe(on_rectangle_change, names="locations")
                     self.map.add_layer(rectangle)
 
                 else:
```